### DISCUSSION TITLE
How to enable asan. tsan, etc for compile_commands

### DISCUSSION THREAD

**Discussion by davemackintosh** [xmake-io/xmake#3630] [Q&A] [answered]
I would like to see all the flags I enable for the target I'm building appear in the compile_commands.json file so my editor can give me the same results as my build but no matter what I try, I can't get the flags for the compiler saved to the compilation database. Either the flags aren't being set at compile time or they aren't considered when generating the compile_commands.json.

**Comment by waruqi** [comment]
please show your xmake.lua configuration.

**Reply by waruqi** [reply]
and if you want to use asan, you can use builtin rules.

```
add_rules("mode.asan", "mode.tsan")
```

then

```
xmake f -m asan
xmake -rv
```

or 

```
xmake project -m asan -k compile_commands
```

**Reply by davemackintosh** [reply]
Ah okay, is there a way to set multiple modes at once for a target? For instance, I'd like to fail tests if there is any asan OR tsan errors as well as any test failures

**Reply by waruqi** [reply, marked as answer]
asan/tsan have different flags and I'm not sure if they can be turned on at the same time. At the moment rule mode, only one can be switched globally.

But you can add multiple flags at the same time yourself with add_cxflags/add_ldflags

**Reply by davemackintosh** [reply]
Okay, yeah that's what I've done for now. Seems to work fine. Thanks for taking the time to answer my question.
